routing/router.go

```diff
@@ -1523,21 +1524,37 @@ func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *route
 	return r.sendPayment(payment, paySession)
 }
 
-// SendToRoute attempts to send a payment as described within the passed
-// LightningPayment through the provided routes. This function is blocking
-// and will return either: when the payment is successful, or all routes
-// have been attempted and resulted in a failed payment. If the payment
-// succeeds, then a non-nil Route will be returned which describes the
-// path the successful payment traversed within the network to reach the
-// destination. Additionally, the payment preimage will also be returned.
-func (r *ChannelRouter) SendToRoute(routes []*route.Route,
-	payment *LightningPayment) ([32]byte, *route.Route, error) {
-
-	paySession := r.missionControl.NewPaymentSessionFromRoutes(
-		routes,
+// SendToRoute attempts to send a payment with the given hash through the
+// provided route. This function is blocking and will return the obtained
+// preimage if the payment is successful or the full error in case of a failure.
+func (r *ChannelRouter) SendToRoute(hash lntypes.Hash, route *route.Route) (
+	lntypes.Preimage, error) {
+
+	log.Tracef("Attempting to send payment %x, using route: %v",
+		hash, newLogClosure(func() string {
+			return spew.Sdump(route)
+		}),
 	)
 
-	return r.sendPayment(payment, paySession)
+	preimage, err := r.sendToSwitch(route, hash)
```

Comment:
This will need to be shuffled around a little bit after reliable payments are in.

Reply:
Yes, but as we already merged the main refactoring of `router.go` some time ago, it won't be much I expect.

Reply:
Reverted to using `sendPayment` for SendToRoute also, to connect better with #2761.